### ISSUE TITLE
[V3] Implement new Concept Exercise: anonymous-functions

### ISSUE THREAD

**Issue by tehsphinx** [exercism/go#1377]
This issue describes how to implement the `anonymous-functions` concept exercise for the Go track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student how to work with `anonymous functions` in Go.

## Learning objectives

- create anonymous function
- create and run anonymous function without assigning to a variable
- scope
- passing in variables vs using variables from the outer scope
- functions as 1st class citizens: parameters and return values

## Out of scope

- loops
- go routines
- defer

## Concepts

- `anonymous-functions`

## Prerequisites

- `functions`
- `types`

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.
